util/signing/integrations/alloy/src/lib.rs

```diff
@@ -0,0 +1,147 @@
+use alloy_consensus::SignableTransaction;
+use alloy_primitives::{hex, Address, ChainId, B256};
+use alloy_signer::{sign_transaction_with_chain_id, Result, Signature as AlloySignature, Signer};
+use k256::ecdsa::{self, VerifyingKey};
+use signer::{
+    cryptography::secp256k1::Secp256k1, Bytes, SignerOperations, SignerError, Signature as MvtSignature, PublicKey};
+use std::fmt;
+
+pub struct HsmSigner {
+    kms: Box<dyn SignerOperations<Secp256k1>+ Sync> ,
+    pubkey: VerifyingKey,
+    address: Address,
+    chain_id: Option<ChainId>,
+}
+
+impl fmt::Debug for HsmSigner {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        f.debug_struct("HsmSigner")
+            .field("chain_id", &self.chain_id)
+            .field("pubkey", &hex::encode(self.pubkey.to_sec1_bytes()))
+            .field("address", &self.address)
+            .finish()
+    }
+}
+
+#[async_trait::async_trait]
+impl alloy_network::TxSigner<AlloySignature> for HsmSigner {
```

Comment:
Have you already worked on the code for how this gets integrated? I think it's still probably easier for us to change the usage s.t. we are producing a raw transaction and signing that rather than tightly coupling with a target API.